test/integration/query/additional.js

```diff
@@ -869,7 +869,118 @@ module.exports = function (knex) {
         });
     });
 
-    it('.timeout(ms, {cancel: true}) should throw error if cancellation cannot acquire connection', async function () {
+    it('.timeout(ms, {cancel: true}) should throw TimeoutError and cancel slow query in transaction', function () {
```

Comment:
This test is a copy of ".timeout(ms, {cancel: true}) should throw TimeoutError and cancel slow query" test with just the query enveloped into a transaction.

You can use this test to confirm that canceling a timeouted query does not work inside a transaction on current `master`. The test will fail there. I've made as little as possible changes to the test, to make sure the cause is the transation.